### PR TITLE
Add read/write adaptor for ndx-pose

### PR DESCRIPTION
**Pull request by roomrys**
### Description
Add read/write adaptor for `ndx-pose`. Only reads/writes predicted instances. Next steps include: 
- means to read/write training data as well 
- integration with `sleap-io` repo

### Types of changes

- [ ] Bugfix
- [x] New feature
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
N/A

### Outside contributors checklist

- [ ] Review the [guidelines for contributing](https://github.com/talmolab/sleap/wiki/Developer-Guide) to this repository
- [ ] Read and sign the [CLA](https://github.com/talmolab/sleap/blob/develop/sleap-cla.pdf) and add yourself to the [authors list](https://github.com/talmolab/sleap/blob/develop/AUTHORS)
- [ ] Make sure you are making a pull request against the **develop** branch (not *main*). Also you should start *your branch* off *develop*
- [ ] Add tests that prove your fix is effective or that your feature works
- [ ] Add necessary documentation (if appropriate)

#### Thank you for contributing to SLEAP!
:heart:

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -26,4 +26,6 @@ seaborn
 pykalman==0.9.5
 segmentation-models==1.0.1
 rich==10.16.1
-certifi>=2017.4.17,<=2021.10.8
\ No newline at end of file
+certifi>=2017.4.17,<=2021.10.8
```
Comment (talmo) [inline requirements.txt:29]: Do we need to version fence this on the upper end? I think some people were having issues with `certifi`

File: sleap/io/dataset.py [hunk 2]
```diff
@@ -2282,6 +2295,7 @@ def numpy(
         video: Optional[Union[Video, int]] = None,
         all_frames: bool = True,
         untracked: bool = False,
+        get_confidence: bool = False,
```
Comment (talmo) [inline sleap/io/dataset.py:2298]: Missing docstring for this, and also I'd name it `return_confidence` instead

File: sleap/io/format/ndx_pose.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+                    name=f"Track{track_idx:03}",
+                    pose_estimation_series=pose_estimation_series,
+                    description=(
+                        f"Estimated positions of {skeleton.name} in video {video_fn} "
+                        f"using SLEAP."
+                    ),
+                    original_videos=[f"{video_fn}"],
+                    labeled_videos=[f"{video_fn}"],
+                    dimensions=np.array([[video.backend.height, video.backend.width]]),
+                    # scorer="DLC_resnet50_openfieldOct30shuffle1_1600",
```
Comment (talmo) [inline sleap/io/format/ndx_pose.py:278]: Delete comment or include metadata from `labels.provenance` if available

**Review by talmo — CHANGES_REQUESTED** [review]

File: sleap/io/format/ndx_pose.py (hunk 3 above)
Comment (roomrys) [reply sleap/io/format/ndx_pose.py:278]: `scorer=str(labels.provenance)`

File: sleap/gui/commands.py [hunk 4]
```diff
@@ -935,11 +975,16 @@ def do_action(cls, context: CommandContext, params: dict):
     @staticmethod
     def ask(context: CommandContext, params: dict) -> bool:
         default_name = context.state["filename"]
-        if default_name:
-            default_name = get_new_version_filename(default_name)
+        if "extension" in params:
+            default_name += f".{params['extension']}"
+            filters = [f"(*.{params['extension']})"]
```
Comment (roomrys) [inline sleap/gui/commands.py:980]: Maybe we can pass in an adaptor class in `params["extensions"]` and then use `params["extensions"].default_ext` for the suffix and `params["extensions"].all_ext` for the filters. Also, in that case, `"extensions"` -> `"adaptor"`.

File: sleap/gui/commands.py [hunk 5]
```diff
@@ -656,6 +664,37 @@ def ask(context: "CommandContext", params: dict) -> bool:
...
+
+        labels = Labels.load_nwb(filename=params["filename"])
+
+        new_window = context.app.__class__()
+        new_window.showMaximized()
+        new_window.loadLabelsObject(labels=labels)
+
+    @staticmethod
+    def ask(context: "CommandContext", params: dict) -> bool:
+        filters = ["NWB (*.nwb)"]
```
Comment (roomrys) [inline sleap/gui/commands.py:679]: Should really reference the `NXDPoseAdaptor` class here and use `NDXAdaptor.name` with `NDXAdaptor.all_exts`

File: sleap/gui/commands.py [hunk 6]
```diff
@@ -908,7 +947,8 @@ def _try_save(context, labels: Labels, filename: str):
         """Helper function which attempts save and handles errors."""
         success = False
         try:
-            Labels.save_file(labels=labels, filename=filename)
+            extension = filename.split(".")[-1]
```
Comment (roomrys) [inline sleap/gui/commands.py:950]: The `PurePath(...).suffix` method returns the extension with `.` still attached

**Review by roomrys — COMMENTED** [review]
Self-review with a few requested changes

File: sleap/gui/commands.py (hunk 6 above)
Comment (roomrys) [reply sleap/gui/commands.py:950]: use pathlib

**Review by talmo — APPROVED** [review]

**Pull request merged**